### ISSUE TITLE
review use of hibernate jar

### ISSUE THREAD

**Issue by pjfanning** [apache/incubator-kie-issues#1685]
The existing license (LGPL) in Hibernate is not compatible with Apache License. I know there is a plan to relicense the Hibernate jar but we can't delay doing an analysis of where we use the Hibernate jar.
It's ok to have a pom dependency on Hibernate jar. The issue is if we ship the jar or classes from it in a binary artifact. I mean any of the files in https://dist.apache.org/repos/dist/release/incubator/kie/10.0.0/

If we do ship the jar in a binary artifact, one workaround would be to omit the jar and provide instructions allowing users ro add the jar themselves.

Would it be possible to do a review before the next RC?

**Comment by tkobayas** [comment]
drools had removed hibernate dependencies by https://github.com/apache/incubator-kie-drools/pull/5764

Could someone please review other repos?

optaplanner -> @lucamolteni 
kogito-runtimes -> @fjtirado ?
kogito-apps -> @pefernan ?
kie-tools (images, operator) -> @tiagobento @ricardozanini ?

Feel free to involve appropriate persons. Thanks!

**Comment by fjtirado** [comment]
@tkobayas @pjfanning 

Kogito runtimes do not use JPA but Kogito Apps heavily use it, both for data_index and audit components. Therefore removing JPA is not a viable option without basically redeveloping both components. 

On top of that, we recently start using Hibernate classes directly to allow querying ProcessInstance.variables property (which has json content) in postgresql. See [here](https://github.com/apache/incubator-kie-kogito-apps/blob/main/data-index/data-index-storage/data-index-storage-postgresql/src/main/java/org/kie/kogito/index/postgresql/ContainsSQLFunction.java#L24-L28)

Therefore we have following options. 
1) Replace Hibernate by OpenJPA, as was made in Drools (where this is not a problem because there is a JPA transaction manager, but not JPA entities) , and study how to achieve the custom function functionality. After a quick study of the OpenJPA documentation, it seems there might be a [way](https://openjpa.apache.org/builds/2.1.1/apache-openjpa/docs/ch13s03.html), but as you see the doc is sparse.
2) Do as  @pjfanning (mainly because Hibernate License will be eventually updated and this will be a temporary fix) proposed. Do not distribute the binary (set the dependency as provided in the pom) and include instructions to add it. 
3) Replace JPA by JDBC (too late in my opinion. Also see latest parragraph) 

I guess we can try OpenJPA as POC to see if 1) is really an alternative to Hibernate for kogito-apps (again, in Drools the situation was kind of different), but @krisv, do we really want to get rid of Hibernate strategically, given that they are goint to change the license and they are a sound name in the industry?  

(Im not a fan on JPA, less of Hibernate, which I always found pretty heavyweight, but curiously enough, since we start grouping events, the JPA cache starts being a performance advantage and the encapsulation provided by JPA Criteria when calculating GraphQL queries makes new DB integration easier)

**Comment by pjfanning** [comment]
> 2. Do as  @pjfanning (mainly because Hibernate License will be eventually updated and this will be a temporary fix) proposed. Do not distribute the binary (set the dependency as provided in the pom) and include instructions to add it.
> 
It's ok for a pom to link to an LGPL jar(not ideal, but ok). The issue is the zips and tar.gz files in https://downloads.apache.org/incubator/kie/10.0.0/ - these cannot include the LGPL jars. They can include scripts to download the LGPL jars if the jar is needed for the build, with instructions detailing the license issue.

I would also like to highlight fat jars (often these are shaded jars) - these are jars that include many classes including 3rd party classes. If you include LGPL classes in these jars, then you have caused a copyleft issue with that fat jar.

The licensing issue is that you can't ship classes or jars that are LGPL but you can link to jars (eg a pom dependency).

**Comment by fjtirado** [comment]
@pjfanning Yes, I got that and that what I proposed (probably not using the proper grammar construct ;)). If the dependency scope is provided, everything will be compiled and tested as it does now but the hibernate jar wont be included in the final binary. 
Users will need to add it manually later or relational dbs  wont work at all (so we will be cheating, in a way)
I want to stress Im assuming this will be temporary till Hibernate folks update their license. 
Also, removing hibernate, from a marketing point of view, is delicated.

**Comment by fjtirado** [comment]
What Im trying to say, beyond the technical point of view and the license legal thing is that this (Hibernate vs OpenJPA)  is a  pivotal decision for the project that Im not empowered to take and should be further discussed with a broad audience (and not in Christmas Eve). From my point of view, Im ok with moving to OpenJPA (although it will indirectly include apache common, which is frequenlty an issue, specially when cleaning up dependencies, because many developers use functions on there, creating a hard dependency on that library that is not really needed)

**Comment by ricardozanini** [comment]
@fjtirado @pjfanning @tkobayas @tiagobento If we use Hibernate only as a link in the POM (`provided`) and don't include it in the apps, we also have to stop publishing all the Kogito and SonataFlow images in our docker/apache repository since these JARs will be included there. Hence, we would distribute it.

In this case, we need to update the CI and add instructions to users to build the images independently. Also, the SonataFlow operator won't be distributed since it depends on the images published on DockerHub.

It will be a huge hit on our platform.

**Comment by pjfanning** [comment]
@ricardozanini maybe you raise a legal issue at https://issues.apache.org/jira/projects/LEGAL/issues

Maybe it would be worth starting with an issue just about the docker images and if they can include the LGPL licensed Hibernate jar.

**Comment by ricardozanini** [comment]
@pjfanning thank you! I opened https://issues.apache.org/jira/browse/LEGAL-695.

**Comment by lucamolteni** [comment]
As for Optaplanner, it uses Hibernate specific types such as `AbstractSingleColumnStandardBasicType` see  https://docs.optaplanner.org/8.34.0.Final/optaplanner-javadoc/org/optaplanner/persistence/jpa/impl/score/AbstractScoreHibernateType.html  so I don't see it possible to replace it with OpenJPA. 

I'm personally fine with removing the jar file from the zip and adding the instructions to readd it, hoping that soon Hibernate will support the Apache license

**Comment by porcelli** [comment]
@pjfanning a quick update as we are planning to release 10.1.0 soon-ish... 

Hibernate is officially relicensing https://in.relation.to/2025/03/14/orm-asl/

Although we won't be able to adopt it overnight, as it has to first being picked up by Quarkus, but this is in the make... just providing here an update.

**Comment by pjfanning** [comment]
Sounds good. Just keep the DISCLAIMER-WIP until we are ready to uptake the Hibernate ASL release.

**Comment by porcelli** [comment]
Absolutely! No plans to change until properly solved!

Thx!

**Comment by ricardozanini** [comment]
Oh this is great news! Thanks for sharing, @porcelli

**Comment by pjfanning** [comment]
Hibernate 7 is released but it only supports Java 17 afaik (and is Apache licensed)

**Comment by porcelli** [comment]
@pjfanning, yes, it's been released and has been included in Quarkus recently. I'm not sure how it's going with Spring Boot side (KIE supports both runtimes).

It's expected that this will still take some time, as we need Hibernate 7 to be incorporated into an LTS version and also properly adopted and integrated into the SB ecosystem.

**Comment by yesamer** [comment]
Hi there, we updated HIbernate to version 7.1.14 in the Quarkus upgrade task https://github.com/apache/incubator-kie-issues/issues/2204.

**Comment by tiagobento** [comment]
So... can we close this!?

**Comment by porcelli** [comment]
Has been addressed in the most recent release
